### PR TITLE
v6.18.4

### PR DESCRIPTION
**Pull request by cherniavskii**
- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version` (`yarn release:version prerelease` for alpha / beta releases).
- [x] Open PR with changes and wait for review and green CI.
- [x] Merge PR once CI is green, and it has been approved.

### Release the packages

- [x] Checkout the last version of the working branch
- [x] Make sure you have the latest dependencies installed: `yarn`.
- [x] Build the packages: `yarn release:build`.
- [x] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [x] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [x] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```bash
git push upstream master:docs-v6 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v6)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v6` deployment.

### Publish GitHub release

- [x] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,42 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 16.18.4
```
Comment (LukasTy) [inline CHANGELOG.md:6]:
```suggestion
## 6.18.4
```

**Review by flaviendelangle — APPROVED** [review]

**Review by michelengelen — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,42 @@
...
+
+- [pickers] Fix `MultiSectionDigitalClock` issues (#11308) @LukasTy
+
+#### `@mui/x-date-pickers-pro@6.18.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@6.18.4`.
+
+### Docs
+
+- [docs] Fix typo (#11323) @cadam11
```
Comment (LukasTy) [inline CHANGELOG.md:40]:
```suggestion
- [docs] Add Pickers FAQ page (#11347) @noraleonte
- [docs] Fix typo (#11323) @cadam11
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,42 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 16.18.4
+
+_Dec 8, 2023_
+
+We'd like to offer a big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:
+
+- 🐞 Bugfixes
```
Comment (LukasTy) [inline CHANGELOG.md:12]:
```suggestion
- 📚 Add Pickers FAQ page
- 🌍 Improve Danish (da-DK) locale on Data Grid
- 🐞 Bugfixes
```

**Review by LukasTy — APPROVED** [review]

**Review by MBilalShafi — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (cherniavskii) [reply CHANGELOG.md]: Oopsie! 😅

**Pull request merged**